### ISSUE TITLE
Scrolling horizontally in grid with pinned columns causes the grid to scroll to the top of the viewport (Safari)

### ISSUE THREAD

**Issue by nw-fid** [angular-ui/ui-grid#5966]
Browser: Safari 10.0.2
UI-Grid: v3.1.1

**Steps to Reproduce**
1. Scroll down vertically so that the top row of the grid is out of the viewport.
2. Scroll horizontally within the main (body) viewport
3. At some point (usually right away) the grid (pinned and body viewports) will be instantly scrolled vertically to the top of the viewport

This is reproducible in Safari on the [Pinning Tutorial](http://ui-grid.info/docs/#/tutorial/203_pinning) page of the ui-grid website.

**Comment by mmgmoh** [comment]
+1

**Comment by mccxiv** [comment]
In Chrome as well, reproducible in the docs

**Comment by mccxiv** [comment]
This is not a clean solution, but I had to do something for our users...

Calling this function in your controller will disable horizontal scrolling events for pinned columns (left-pinned only). It solves this scrolling bug but probably adds usability issues. It needs `$scope` and `$interval` available, as well as ES6.

```javascript
/**
 * Issue: https://github.com/angular-ui/ui-grid/issues/5966
 * © 2016 Andrea Stella, MIT license.
 */
function fixScrollingBug() {
  const selectors = [
    '.ui-grid-render-container-left .ui-grid-cell-contents:not(.scroll-fix)',
    '.ui-grid-render-container-left .ui-grid-cell:not(.scroll-fix)'
  ].join(', ');
  const fixInterval = $interval(fix, 500);

  $scope.$on('$destroy', () => {
    clearTimeout(fixInterval);
    const elements = [...document.querySelectorAll(selectors)];
    elements.forEach(element => {
      element.removeEventListener('wheel', preventHorScroll);
    });
  });

  function fix() {
    const elements = [...document.querySelectorAll(selectors)];
    if (!elements.length) return;
    elements.forEach(element => {
      element.addEventListener('wheel', preventHorScroll);
      element.classList.add('scroll-fix');
    });
  }

  function preventHorScroll(e) {
    if (e.deltaX !== 0) {
      e.preventDefault();
      e.stopPropagation();
    }
  }
}
```

**Comment by mmgmoh** [comment]
@mccxiv Thank you for the proposed solution. I tried implementing it on my safari problem with ui-grid, unfortunately it does not work. The scrolling is still very laggy and ui-grid still jumps to the top of the grid when quickly doing horizontal scroll.

**Comment by mmgmoh** [comment]
After reviewing the code in the ui-grid javascript file, it seems like the error is coming from the scrollHandler function specifically from the following line: `var newScrollTop = $elm[0].scrollTop;`. 
Somehow when we reproduce the horizontal scroll bug, some additional scroll events are fired and in some of those events $elm[0].scrollTop decreases in Safari as if we scrolled back to the top.

**Comment by notomorow** [comment]
This is not ui-grid's bug, it's because the onscroll event behave  different on webkit browser

**Comment by mmgmoh** [comment]
@notomorow  yes Webkit behaves differently but Chrome also uses Webkit and works fine. I am lead to believe the bug maybe due to Safari's rubber band scrolling feature.

**Comment by notomorow** [comment]
@mmgmoh what's your chrome's version?

**Comment by Fraraven** [comment]
has anyone found a solution for this?
Thanks

**Comment by orneryd** [comment]
Thanks for contributing! This repository has been transferred, and we're preparing the new 5.x release. We're closing older issues and pull requests as part of the reset. Please join us on Discord to follow the work or continue the conversation: https://discord.gg/Baz4w8ZWN
